consensus/consensus_v2.go

```diff
@@ -184,7 +186,7 @@ func (consensus *Consensus) finalCommit() {
 	// Note: leader already sent 67% commit in preCommit. The 100% commit won't be sent immediately
 	// to save network traffic. It will only be sent in retry if consensus doesn't move forward.
 	// Or if the leader is changed for next block, the 100% committed sig will be sent to the next leader immediately.
-	if !consensus.isLeader() || block.IsLastBlockInEpoch() {
+	if !isLeader || block.IsLastBlockInEpoch() {
```

Comment:
since isLeader() is already in consensus, why passing isLeader to the function now ?

Reply:
Function runs in goroutine and sometimes `consensus.isLeader()` returns `false` where should be `true`